src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const operationsResult = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+      case 'removeProperties':
+        action.keysToRemove.map(key => delete stateCopy[key]);
```

Comment:
I see your point, but you are using `map` method incorrectly - it is intended to map the array onto some other data and generate a different array of the same size.
But `Array` has a different method that is perfect here ;)